### ISSUE TITLE
Suggestion: GitHub bot to auto update and render files

### ISSUE THREAD

**Issue by Milo123459** [excalidraw/excalidraw#3498]
Hi!

I had a suggestion, a github bot, that commits to your project, automatically updating the excalidraw file, if it exists, and then updating a rendered photo. 

So lets say this:

I draw a box with hello in it, I link it up to the project, you give it a branch to go on, and then it adds those files to the branch.

Now lets say I change the text to say how are you

The github bot would automatically update those files on the github repo

Pointless but cool for projects that like to have automated sketch ups, and would be very useful for some of my projects.

Keep up the great work!

**Comment by dwelle** [comment]
Actually, something similar was planned and I meant to work on this but stuff got in the way.

/cc @gr2m what do you think about the bot idea? Is it a feasible alternative to the extension (after all, how many people will have it installed)?

**Comment by Milo123459** [comment]
I mean a server for a github bot doesn't need to have a lot of resources, especially for something like this, everything is down on excalidraw's servers for the website.

**Comment by gr2m** [comment]
See https://github.com/excalidraw/excalidraw/discussions/2716 for the previous discussions on that topic, it was partly implemented in https://github.com/excalidraw/excalidraw/pull/2726 (the open excalidraw file by URL part). What's left is to write an updated Excalidraw file back to GitHub. I'm not sure if a server part is needed for that, it's been a while since I wrapped by head around it. 

Milo, can you check if what we discussed would cover your use case, and if you have anything to add?

**Comment by Milo123459** [comment]
I shall have a look!

**Comment by Milo123459** [comment]
Yes, that covers everything I imagine. I think there needs to be a way to import from a hosted provider to make changes. Ie, import from github then provide a repo URL with a branch and path to the file, you can then authenticate with github, and if you have perms to commit on the repo, commit on the repo, if not, fork, make new branch, and create a pr.

**Comment by gr2m** [comment]
@dwelle maybe the next step would be to create an actionable issue as follow up to this one and the discussion at #2716 which what still needs to be done, and how it would work?

If I understand it correctly, excalidraw.com is deployed to Vercel. Would you mind adding a stateless serverless function to the deployment? Then we could implement an OAuth login. If yes, then I'd suggest to work on this with a separate repository / vercel deployment for now, for quicker iteration. Then once we are happy with how things work, add the function to the excalidraw.com deployment. 

I think the function would only need to take care of the OAuth flow, everything else could work from the browser app. 

If the source file on GitHUb is an SVG or PNG, can you generate both file and embed the source JSON into the files in the browser? Or is that something that can only be done serverside right now?

**Comment by Milo123459** [comment]
Oh by the way, I'm happy to help work on this. I've personally used https://next-auth.js.org/ but since excalidraw doesn't use next (and I'm sure it is compatible) it won't work. 

If you can let me know how I can help, that'd be great!
